src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,326 @@
+			if(uri is null)
+			{
+				throw new ArgumentException("ToastNotifier:ConvertToastTextToString, toastText cannot be converted to Uri");
+			}
+			// "When using the "ms-resource" prefix, the string identifier is referenced in the app's Resources.resw file."
+			if (uri.Scheme == "ms-resource")
+			{
+				return toasttext;
+			}
+
+			var resName = uri.SchemeSpecificPart;
+			if (resName is null)
+			{
+				throw new ArgumentException("ToastNotifier:ConvertToastTextToString, empty SchemeSpecificPart (within Scheme ms-resource)");
+			}
+			toasttext = resName; 
+			var retVal = Windows.ApplicationModel.Resources.ResourceLoader.GetForCurrentView()?.GetString(toasttext);
+			if (retVal is null)
+			{   // we have no such string
+				return toasttext;
+			}
+			return retVal;
+		}
+
+		private Android.App.Notification.Builder GetToastBuilder()
+		{
+			// if API < 11, then another version...
+			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+			{
+				return new Android.App.Notification.Builder(Android.App.Application.Context, _channelID);
+			}
+
+#pragma warning disable CS0618 // Type or member is obsolete - compilation is for newer SDK, but runtime is on older SDK
+			return new Android.App.Notification.Builder(Android.App.Application.Context);
+#pragma warning restore CS0618
+
+		}
+
+		private int GetIconId(string packageName)
+		{
+			var info = Android.App.Application.Context?.ApplicationContext?.PackageManager?.GetApplicationInfo(packageName, 0);
+
+			int iconId;
+			if (info != null)
+			{
+				iconId = info.Icon;
+			}
+			else
+			{
+				// try another method - using more resources (CPU and memory), as Intent is involved
+				var context = Android.App.Application.Context;
+				if (context?.PackageName is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, Context.PackageName is null");
+				}
+
+				var intent = context.PackageManager?.GetLaunchIntentForPackage(context.PackageName);
+				if (intent is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, GetLaunchIntentForPackage is null");
+				}
+
+				var resolveInfo = context.PackageManager?.ResolveActivity(intent, Android.Content.PM.PackageInfoFlags.MatchDefaultOnly);
+				if (resolveInfo is null)
+				{
+					throw new InvalidOperationException("ToastNotifier:GetIconId, ResolveActivity is null");
+				}
+				iconId = resolveInfo.IconResource;
+			}
+
+			// iconId == null means error, but we throw exception in Show method - logic would be more readable
+			return iconId;
+
+		}
+
+		private void SetToastTexts(Android.App.Notification.Builder builder, XmlDocument xmlDoc)
+		{
+			string toastTitleText = "";
+			StringBuilder toastText = new StringBuilder("");
+
+			var childToast = xmlDoc.GetElementsByTagName("text");
+			if ((childToast != null) && childToast.Count > 0)
+			{
+				// first text - bigger text (title)
+				toastTitleText = ConvertToastTextToString(childToast![0]!.InnerText);

```

Comment:
Try to avoid nullable suppressions. If the compiler tells you it can be null, ensure that it cannot:

```suggestion
                                if (childToast[0] is {} firstChild)
                                {  
				    toastTitleText = ConvertToastTextToString(firstChild.InnerText);
				}

```

Reply:
But it cannot. As line 119 says: `if ((childToast != null) && childToast.Count > 0)` , so `childToast[0]` has to be not null here.

Reply:
Can this conversation be marked as resolved?

Reply:
Ok, adding one more test.